custom/icds_reports/reports/awc_reports.py

```diff
@@ -1171,11 +1171,9 @@ def get_pregnant_details(case_id, awc_id):
             [],
         ],
     }
-    current_trimester = 1
-    current_record = 0
     for row_data in data:
-        if row_data['trimester'] >= current_trimester:
-            config['data'][row_data['trimester'] - 1].append(dict(
+        config['data'][row_data['trimester'] - 1].append(
```

Comment:
There will be one record in this table for every month, once the case exists, but we only want the months in which a home visit occurred. I think you will have to filter here for months in which home visit date matches the month.